### PR TITLE
Sensor reading exercise skeleton & exercise part 1 & 2

### PR DESCRIPTION
**Pull request by Mirabellensaft**

### CODE REVIEW

**Comment by Mirabellensaft** [comment]
The Exercise does not contain a step by step solution yet, I want to leave that out for now, and add it if we are within the specified time. People can still work with the complete solution code for help

**Comment by Mirabellensaft** [comment]
I removed the exercise/solution hustle to avoid having to update resources in 2 places. The solutions are now in the example folder, and can be run with adding --example to the command. Book update follows.

File: book/src/04_3_1_i2c.md [hunk 1]
```diff
@@ -1,21 +1,132 @@
...
 
+### Part 1: Reading Temperature & Humidity
 
-TODO `i2c-driver/exercise/src/main.rs` contains working code. Running it without modifying the driver file will yield errors. 
+Create an instance of the of the Humidity sensor SHTC3 and read and print the values for humidity and temperature every 600 ms.
 
-`i2c-driver/solution/src/imc42670p.rs` provides a solution to the task. 
 
 
+`i2c-sensor-reading/examples/part_1.rs` contains a working solution of Part 1.
```
Comment (BriocheBerlin) [inline book/src/04_3_1_i2c.md:23]: Maybe add how to run the solution?

File: book/src/04_3_1_i2c.md [hunk 2]
```diff
@@ -1,21 +1,132 @@
...
+* There are methods that turn the sensor values into the desired unit. 
+
+### Part 2: Reading Accelerometer data. 
+
+Using a bus manager, implement the second sensor. Read out its values and print the values from both sensors. 
+
+
+Continue with your code from Part 1. Alternatively you can start with the provided partial solution of Part 1: `i2c-sensor-reading/examples/part_1.rs`.
+
+`i2c-sensor-reading/examples/part_2.rs` contains a working solution of Part 1.` contains a working solution of Part 2. You can consult it if you need help.
```
Comment (BriocheBerlin) [inline book/src/04_3_1_i2c.md:80]: Seems like these sentences are not correct / either the first or second sentence should be removed.

File: book/src/04_3_1_i2c.md [hunk 3]
```diff
@@ -1,21 +1,132 @@
...
+```
+ 
+✅ Define two pins, one as SDA and one as SCL. 
+
+| Signal     | GPIO        |
+|------------|-------------|
+| SDA        | GPIO10      |
+| SCL        | GPIO8       |
+
+✅ Create an Instance of the I²C peripheral with the help of the documentation.
```
Comment (BriocheBerlin) [inline book/src/04_3_1_i2c.md:42]: Maybe some people won't know where to look exactly. One possibility would be to mention `esp_idf_hal`.

File: book/src/04_3_1_i2c.md (hunk 2 above)
Comment (Mirabellensaft) [reply book/src/04_3_1_i2c.md:80]: Ah yes I understand what you mean.

File: book/src/04_3_1_i2c.md (hunk 3 above)
Comment (Mirabellensaft) [reply book/src/04_3_1_i2c.md:42]: its your own generated documentation!

File: advanced/i2c-sensor-reading/examples/part_2.rs [hunk 4]
```diff
@@ -25,49 +25,52 @@ fn main() -> anyhow::Result<()>  {
...
     let sda = peripherals.pins.gpio10;
     let scl = peripherals.pins.gpio8;
-    // If you are using an ESP32-C3-DevKitC-02, change to:
-    // let sda = peripherals.pins.gpio4;
-    // let scl = peripherals.pins.gpio5;
 
     let i2c = Master::<I2C0, _, _>::new(
         peripherals.i2c0,
         MasterPins { sda, scl },
         <MasterConfig as Default>::default().baudrate(400.kHz().into()),
```
Comment (BriocheBerlin) [inline advanced/i2c-sensor-reading/examples/part_2.rs:32]: Maybe the baudrate should get mentioned somewhere.

**Comment by BriocheBerlin** [comment]
> I removed the exercise/solution hustle to avoid having to update resources in 2 places. The solutions are now in the example folder, and can be run with adding --example to the command. Book update follows.

Best to delete the solution folder entirely.

**Pull request merged**